### ISSUE TITLE
merging into mxnet repo

### ISSUE THREAD

**Issue by hjk41** [dmlc/MXNet.cpp#71]
I propose to merge mxnet.cpp into mxnet project.
1. It will attract more attention to mxnet.cpp.
2. It will enable integrated test with mxnet, so that future checkins to mxnet will not break mxnet.cpp

**Comment by hjk41** [comment]
@lx75249 @zhangchen-qinyinghua

**Comment by conopt** [comment]
The most convinient way is to add a subdirectory (cpp-package) in mxnet's root, like other language bindings (R and scala). In this way, the main work is test integration. (I guess we can also add some hook for regenerating op.h when mxnet is updated? I'm not sure if it's possible in github)

**Comment by hjk41** [comment]
I agree with the subdirectory approach. As for the update hook, I guess we can hook it, but not sure if we can execute python script inside the hook.
Anyway, let's merge it first, and then try to figure out the hook thing.
@lx75249 Could you create a PR to mxnet with the subdirectory?

**Comment by conopt** [comment]
OK. I will do it.

**Comment by qiyuangong** [comment]
I think this should be done ASAP. Because this repo is far behind other interfaces, e.g, Python. Meanwhile, it  is really important for industrial envo (e.g, data center), where Python and other dependencies are normally not installed.

**Comment by zhangchen-qinyinghua** [comment]
I agree :)
